models/review.go

```diff
@@ -74,24 +74,48 @@ type Review struct {
 }
 
 func (r *Review) loadCodeComments(e Engine) (err error) {
+	if err = r.loadIssue(e); err != nil {
+		return err
+	}
 	if r.CodeComments == nil {
 		r.CodeComments, err = fetchCodeCommentsByReview(e, r.Issue, nil, r)
 	}
 	return
 }
 
+// GetCodeCommentsCount return count of CodeComments a Review has
+func (r *Review) GetCodeCommentsCount() int {
+	opts := FindCommentsOptions{
+		Type:     CommentTypeCode,
+		IssueID:  r.IssueID,
+		ReviewID: r.ID,
+	}
+	conds := opts.toConds()
+	if r.ID == 0 {
+		conds = conds.And(builder.Eq{"invalidated": false})
+	}
+
+	count, err := x.Where(conds).Count(new(Comment))
+	if err != nil {
+		return 0
+	}
+	return int(count)
+}
+
 // LoadCodeComments loads CodeComments
 func (r *Review) LoadCodeComments() error {
 	return r.loadCodeComments(x)
 }
 
 func (r *Review) loadIssue(e Engine) (err error) {
-	r.Issue, err = getIssueByID(e, r.IssueID)
+	if r.Issue == nil {
```

Comment:
this should probably be other way around.
```go
	if r.Issue != nil {
		return
	}
	r.Issue, err = getIssueByID(e, r.IssueID)
```

Reply:
code formation ...

I think it wont make a difference but I'll change it